docs/specs/code/SIGNING.md

```diff
@@ -0,0 +1,424 @@
+
+# Related specifications
+
+### Requirement Levels [[RFC 2119](https://tools.ietf.org/html/rfc2119)]
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+### Concise Binary Object Representation (CBOR) [[RFC 7049](https://tools.ietf.org/html/rfc7049)]
+
+CBOR is a way to serialize structured data in a more compact way than what is allowed by JSON. IT is widely used across the Cardano ecosystem and so we use it to encode the data for this specification.
+
+There is currently no standard CBOR grammar available for use by specifications. The CBOR structures are therefore described in prose.
+
+| Prose | description           | Major type | value |
+|-------|-----------------------|------------|-------|
+| uint  | unsigned int          |      0     |   -   |
+| nint  | negative int          |      1     |   -   |
+| bstr  | byte string           |      2     |   -   |
+| tstr  | UTF8 string           |      3     |   -   |
+| array | arbitrary size array  |      4     |   -   |
+| map   | arbitrary size map    |      5     |   -   |
+| tag   | semantic tag for data |      6     |   -   |
+| false |                       |      7     |   20  |
+| true  |                       |      7     |   21  |
+| nil   |                       |      7     |   22  |
+
+Some additional shorthand notations is used:
+
+- any - non-specific value that permits all CBOR values to be placed here.
+- int - an unsigned integer or a negative integer.
+- FOO / BAR - indicates that either FOO or BAR can appear here
+- bool - false / true
+- label - int / tstr
+- [+ FOO] - type FOO appears one or more times in an array
+- [* FOO] - type FOO appears zero or more times in an array
+- [? FOO] - type FOO appears 0 or 1 time in an array
+
+### *CBOR Object Signing and Encryption (COSE)* [[RFC 8152](https://tools.ietf.org/html/rfc8152)] 
+
+This is a standard for how to use CBOR for message signing. It is based on the JSON equivalent *JSON Object Signing and Encryption* [RFC 7520](https://tools.ietf.org/html/rfc7520). 
+
+We base our construction on COSE because all Cardano libraries already depend on CBOR due to its use in the base protocol (which means we don't need to introduce a new library). It is also more compact, which is useful in case data generated by this standard ever needs to be stored on-chain.
+
+Note that we do NOT follow the COSE specification exactly for the following reason:
+
+- We need to restrict the number of algorithms used to only ones that would already be required to provide base Cardano functionality (see E1).
+- We need to enforce the usage of a digest if the input string is not ASCII (see E3)
```

Comment:
Probably can remove this line since I don't think it needs any changes beyond just a new header element